### ISSUE TITLE
[object_store] Support generating and using signed upload URLs

### ISSUE THREAD

**Issue by carols10cents** [apache/arrow-rs#4763]
IOx needs the ability to ask object storage for a signed URL that can be then used for a PUT request to upload a file to a particular path.

This is related to, but slightly different from https://github.com/apache/arrow-rs/issues/3027, which is for generating signed URLs for GET requests to read particular files. (it might be easy enough to add that functionality when this ticket is done.

**Comment by tustvold** [comment]
I wonder if we might add this functionality to [AwsAuthorizer](https://docs.rs/object_store/latest/object_store/aws/struct.AwsAuthorizer.html). Perhaps as a sign method to go alongside the authorize?

Eventually we could potentially extract a trait, and provide similar abstractions for all remote stores. I'm mainly just thinking it probably doesn't make sense to bundle this into the ObjectStore trait when many stores it won't make sense for.

**Comment by carols10cents** [comment]
Sure, sounds good!

**Comment by alamb** [comment]
@carols10cents  and @stuartcarnie -- is this feature needed for anything "near term" in IOx? @tustvold  and I are trying to make sure it is prioritized correctly.

**Comment by carols10cents** [comment]
@alamb Yes, it's needed for bulk migration: https://github.com/influxdata/idpe/issues/18069

**Comment by alamb** [comment]
@tustvold has put this item on his list

**Comment by alamb** [comment]
@tustvold has put this item on his list

**Comment by tustvold** [comment]
Potentially if you don't get to it first, my list is depressingly long... :sweat_smile:

**Comment by tustvold** [comment]
`label_issue.py` automatically added labels {'object-store'} from #4876
